plugins/destination/gremlin/client/read.go

```diff
@@ -28,27 +30,36 @@ func (c *Client) Read(ctx context.Context, table *schema.Table, sourceName strin
 	}
 	defer rs.Close()
 
+	//var records []arrow.Record
 	for row := range rs.Channel() {
 		m := row.Data.(map[any]any)
 		for _, rowCols := range m {
 			rowData := rowCols.(map[any]any)
-
-			madeRow := make([]any, len(table.Columns))
-			for i, col := range table.Columns {
-				if rowData[col.Name] == nil {
-					continue
-				}
-
-				data := rowData[col.Name].([]any)
-				if l := len(data); l == 1 {
-					madeRow[i] = data[0]
-				} else if l > 1 {
-					return fmt.Errorf("expected 1 value for %v, got %v", col.Name, l)
-				}
+			rec, err := reverseTransformer(table, rowData)
+			if err != nil {
+				return err
 			}
-			res <- madeRow
+			//records = append(records, rec)
+			res <- rec
 		}
 	}
 
+	//syncTimeIndex := table.FieldIndices(schema.CqSyncTimeColumn.Name)[0]
```

Comment:
This would also work rather than sorting things in the SDK... but why should every dest plugin have to deal with the same issue, and also hardcoding internal cq-column types is... bad